packages/tracing/src/hubextensions.ts

```diff
@@ -61,25 +63,27 @@ function sample<T extends Transaction>(hub: Hub, transaction: T, samplingContext
   let sampleRate;
   if (typeof options.tracesSampler === 'function') {
     sampleRate = options.tracesSampler(samplingContext);
-    // cast the rate to a number first in case it's a boolean
-    transaction.tags = {
-      ...transaction.tags,
-      __sentry_samplingMethod: TransactionSamplingMethod.Sampler,
-      // TODO kmclb - once tag types are loosened, don't need to cast to string here
-      __sentry_sampleRate: String(Number(sampleRate)),
-    };
+    transaction.setMetadata({
+      transactionSampling: {
+        method: TransactionSamplingMethod.Sampler,
+        // cast to number in case it's a boolean
+        rate: Number(sampleRate),
```

Comment:
@jan-auer Once consequence of this change is that rates will come through as numbers rather than strings. (When I did the original PR, tags had to be strings, which is why I stringified the value in the first place.) LMK if that's not good from a relay perspective and I can turn them back into strings.

Reply:
Is this what you put into the `sample_rates` item header? The schema in there is:

```ts
type SampleRate = {
  name: String,
  rate: Number,
}
```